From 3d88f8c08b477bd96478db2259c8b4a95e94bac7 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Sun, 23 Feb 2025 22:36:36 +0100
Subject: [PATCH] Add option to install hera

---
 CMakeLists.txt | 38 ++++++++++++++++++++++----------------
 1 file changed, 22 insertions(+), 16 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5bd6812..925c444 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,6 +38,8 @@ option(XEUS_R_USE_SHARED_XEUS "Link xr  with the xeus shared library (instead of
 option(XEUS_R_USE_SHARED_XEUS_R "Link xr  with the xeus-r shared library (instead of the static library)" ON)
 option(XEUS_R_EMSCRIPTEN_WASM_BUILD "Build for wasm with emscripten" OFF)
 
+option(INSTALL_HERA "Install the hera R package" ON)
+
 if(EMSCRIPTEN)
     add_compile_definitions(XEUS_R_EMSCRIPTEN_WASM_BUILD)
     message("Build with emscripten")
@@ -74,24 +76,28 @@ message(STATUS "R_SCRIPT_COMMAND = ${R_SCRIPT_COMMAND}")
 # Install R package hera
 # ======================
 
-message(STATUS "Installing R 📦 hera to ${R_HOME}/library")
-if(EMSCRIPTEN)
-    execute_process(COMMAND ${R_COMMAND} CMD INSTALL
-        --preclean --no-staged-install --no-byte-compile --no-test-load
-        --build ../hera --library=${R_HOME}/library
-        RESULT_VARIABLE hera_result
-    )
+if(INSTALL_HERA)
 
-else()
-    execute_process(COMMAND ${R_COMMAND} CMD INSTALL
-        --preclean --no-staged-install --build ../hera
-        RESULT_VARIABLE hera_result
-    )
-    execute_process(COMMAND ${R_SCRIPT_COMMAND} -e "writeLines(paste('Installed 📦 hera version', as.character(packageVersion('hera'))))")
-endif()
+    message(STATUS "Installing R 📦 hera to ${R_HOME}/library")
+    if(EMSCRIPTEN)
+        execute_process(COMMAND ${R_COMMAND} CMD INSTALL
+            --preclean --no-staged-install --no-byte-compile --no-test-load
+            --build ../hera --library=${R_HOME}/library
+            RESULT_VARIABLE hera_result
+        )
+
+    else()
+        execute_process(COMMAND ${R_COMMAND} CMD INSTALL
+            --preclean --no-staged-install --build ../hera
+            RESULT_VARIABLE hera_result
+        )
+        execute_process(COMMAND ${R_SCRIPT_COMMAND} -e "writeLines(paste('Installed 📦 hera version', as.character(packageVersion('hera'))))")
+    endif()
+
+    if(NOT hera_result EQUAL "0")
+        message(FATAL_ERROR "Hera installation failed with return code ${hera_result}")
+    endif()
 
-if(NOT hera_result EQUAL "0")
-    message(FATAL_ERROR "Hera installation failed with return code ${hera_result}")
 endif()
 
 # Configuration
